solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java

```diff
@@ -1208,6 +1208,7 @@ private void setupPolling(String intervalStr) {
         log.info("Poll disabled");
         return;
       }
+      ExecutorUtil.setServerThreadFlag(true); // so PKI auth works
```

Comment:
The existing two `ExecutorUtil.setServerThreadFlag(true/BOOLEAN.true)` calls have a matching `setServerThreadFlag(false/null)` afterwards. Any thoughts on having the same here? Perhaps it's not necessary but having it even then (or a comment about why it's not necessary to unset) would help with code reading?

Reply:
I don't think it's needed to unset this `ThreadLocal` given it's a single thread running in a dedicated scheduled executor, so the thread is not shared for other purposes. It doesn't hurt to unset either if we think that makes the code easier to read.

Reply:
I agree with @cpoerschke - unset it in the finally block. I don't want somebody else to later come in and accidentally re-use this thread pool and get surprised with a RCE.